src/shared/bpf-program.h

```diff
@@ -40,3 +41,24 @@ int bpf_map_update_element(int fd, const void *key, void *value);
 int bpf_map_lookup_element(int fd, const void *key, void *value);
 
 DEFINE_TRIVIAL_CLEANUP_FUNC(BPFProgram*, bpf_program_unref);
+
+typedef struct BPFProgramItem BPFProgramItem;
+
+struct BPFProgramItem {
+        BPFProgram *prog;
+        LIST_FIELDS(BPFProgramItem, items);
+};
+
+
+int bpf_program_item_new(BPFProgramItem **ret);
+
+typedef struct BPFProgramList BPFProgramList;
+
+struct BPFProgramList {
+        unsigned n_ref;
+        LIST_HEAD(BPFProgramItem, list);
+};
```

Comment:
hmm, why do we need two new types for that? i mean, if we want to allow multiple of these then it might be simpler to just use a `Set`. Or maybe even a realloc()able array, after all this is not going to be that many new things.

Moreover, it appears to me that having this ref-counted as a whole is unnecessary. It makes more sense to me, to ref count only the programs themselves as we do, but inc their ref as soon as we add them two any `Set`... I mean, the activation of all bpf progs is not going to be atomic anyway but step-by-step...